### PR TITLE
rgw: use a namespace for rgw reshard pool for upgrades as well

### PR DESCRIPTION
**Pull request by theanalyst**
We were previously (for dev & RC1) creating a new pool for bucket index
resharding during upgrades, moving to a namespace similar to new deployments since this
will avoid the creation of a new pool, also clarify the log message if
we do encounter the situation where the pool can't be accessed.

However there might be possible misconfigurations seen when upgrading
from luminous dev/RC1 releases to RC2 where a different pool for reshard
will still be used from the previous upgrade. Also need clarification
whether we need to bump up the zone params struct version number from 10
to 11

Fixes: http://tracker.ceph.com/issues/20289

### CODE REVIEW

**Comment by theanalyst** [comment]
@cbodley @oritwas reraising https://github.com/ceph/ceph/pull/16193 after incorporating the namsepace comment from tracker.. is this the way to proceed?

File: src/rgw/rgw_json_enc.cc [hunk 1]
```diff
@@ -904,6 +904,7 @@ void RGWZoneParams::dump(Formatter *f) const
   encode_json("log_pool", log_pool, f);
   encode_json("intent_log_pool", intent_log_pool, f);
   encode_json("usage_log_pool", usage_log_pool, f);
+  encode_json("reshard_pool", reshard_pool, f);
```
Comment (cbodley) [inline src/rgw/rgw_json_enc.cc]: it looks like `reshard_pool` is missing from `RGWZoneParams::decode_json()` too

Comment (theanalyst) [inline src/rgw/rgw_json_enc.cc]: thanks! updating

**Comment by cbodley** [comment]
> Also need clarification whether we need to bump up the zone params struct version number from 10 to 11

i don't think that's necessary, since you're just changing the default value when decoding an older entry

File: src/rgw/rgw_json_enc.cc (hunk 1 above)
Comment (theanalyst) [inline src/rgw/rgw_json_enc.cc]: updated

**Comment by theanalyst** [comment]
@cbodley updated

**Comment by yuriw** [comment]
http://pulpito.ceph.com/yuriw-2017-07-18_19:38:33-rgw-wip-yuri-testing2_2017_7_19-distro-basic-smithi/

**Pull request merged**